interfaces/ldap/v0/schema.py

```diff
@@ -0,0 +1,59 @@
+"""This file defines the schemas for the provider and requirer sides of this
+relation interface.
+
+It must expose two interfaces.schema_base.DataBagSchema subclasses called:
+- ProviderSchema
+- RequirerSchema
+"""
+
+from typing import Optional
+
+from interface_tester.schema_base import DataBagSchema
+from pydantic import AnyUrl, BaseModel, Field
+
+
+class LdapProviderData(BaseModel):
+    ldap_url: AnyUrl = Field(
+        description="The LDAP URL",
+        example="ldap://ldap.canonical.com:3893",
+    )
+    base_dn: str = Field(
+        description="The base entry as the starting point for LDAP search "
+                    "operation",
+        example="dc=canonical,dc=com",
+    )
+    bind_dn: str = Field(
+        description="The DN of the bind account",
+        example="cn=admin,ou=engineering,dc=canonical,dc=com",
+    )
+    bind_password: str = Field(
```

Comment:
```suggestion
    bind_dn: str = Field(
        description="The DN of the bind account",
        example="cn=admin,ou=engineering,dc=canonical,dc=com",
    )
    bind_password: str = Field(
```

consider making these properties of a `bind` field.
also consider renaming `DN`s `distinguished_name`, as we're not in a shortage of characters and it would help with comprehension at a glance.

Reply:
IMHO, `dn` is fair for users to understand what the field is. `dn` is widely used in different LDAP client configurations, e.g. [SSSD](https://linux.die.net/man/5/sssd-ldap) and [Postgresql](https://www.postgresql.org/docs/current/auth-ldap.html). Charm developers need to learn a bit of LDAP when developing the client charmed operator.

If this is OK, I think it's not necessary to make it nested since the fields are not going to expand for `bind` operation in a foreseeable period. And the stakeholders have agreed on the databag's structure.